commands/run.go

```diff
@@ -0,0 +1,209 @@
+// directory, in `$HOME/.redalert/` (`%APPDATA%` instead of `$HOME` on windows.)
+// and finally in `/etc/redalert/` (`C:\redalert` on windows).
+//
+// It takes the following flags:
+//
+//  - `--quiet` only report test failures.
+//  - `--jobs $VALUE` specify the number of parallel tests to run. Default to # of cores.
+//  - `--test $TEST_NAME` specify a test by name to run, can be provided multiple times.
+//  - `--suite $SUITE_NAME` which test suite to run, an [alias](#aliases) can be provided as the suite name.
+//    Can be provided multiple times. There is a default "all" alias which matches
+//    all tests and is used if `--suite` is not provided.
+//  - `--file $FILE_PATH` in lieu of searching in directories you can specify a file using this flag.
+//    If provided the aliases.(yml|yaml) file will not be looked for but aliases in the file will still be loaded.
+//  - `--output $FORMAT` specify the output format. Valid values: text, json, csv. Default: text
+//
+// Run will attempt to load an additional file `aliases.(yml|yaml)` which
+// specifies the available [aliases](#aliases). If not found aliases will be
+// looked for as a key in the `tests.(yml|yaml)` according to the
+// [aliases](#aliases) section.
+//
+// Run will run tests in parallel.
+var Run = &cobra.Command{
+	Use:   "run",
+	Short: "Run tests against this system.",
+	Run: func(cmd *cobra.Command, args []string) {
+		var err error
+		var testsFile testfile.TestFile
+
+		if fileFlag == "" {
+			testsFile, err = loadTestFile(findTestFile())
+		} else {
+			testsFile, err = loadTestFile(fileFlag)
+		}
+
+		if err != nil {
+			fmt.Println("ERROR:", err)
+			os.Exit(1)
+		}
+
+		results := map[string]error{}
+
+		for _, suite := range suites {
+			tests := testsFile.TestsToRun(suite)
+			loadedChecks, err := checks.LoadChecks(tests)
+			if err != nil {
+				fmt.Println("ERROR: Unable to load checks:", err)
+				continue
+			}
+
+			checksToRun := make(chan checks.CheckToRun, len(loadedChecks))
+			checkResults := make(chan checkResult, len(loadedChecks))
+
+			for i := 0; i < runtime.NumCPU(); i++ {
+				go runCheck(checksToRun, checkResults)
+			}
+
+			for _, c := range loadedChecks {
+				checksToRun <- c
+			}
+
+			for i := 0; i < len(loadedChecks); i++ {
+				result := <-checkResults
+				results[result.Name] = result.Result
+			}
+
+			close(checksToRun)
+			close(checkResults)
+		}
+
+		exitCode := 0
+
+		for name, result := range results {
+			if result == nil {
+				fmt.Printf("%s: SUCCESS\n", name)
+				continue
+			}
+
+			fmt.Printf("%s: FAILURE\n%s\n", name, result)
+			exitCode = 1
+		}
+
+		os.Exit(exitCode)
+	},
+}
+
+type checkResult struct {
+	Name   string
+	Result error
+}
+
+func runCheck(incomingChecks chan checks.CheckToRun, results chan checkResult) {
+	for check := range incomingChecks {
+		results <- checkResult{
+			Name:   check.Name,
+			Result: check.Check(),
+		}
+	}
+}
+
+func findTestFile() string {
+	possiblePaths := []string{
```

Comment:
I don't think these match the comment above...

- The comment reads to me as looking for /etc/redalert/redalert.(yml|yaml) with an extra "/redalert/" directory component vs /etc/redalert.(yml|yaml)
- Comment reads to me as looking for c:\redalert\redalert.(yml|yaml) not c:\redalert.(yml|yaml)
- Comment says ".redalert" for HOME/APPDATA
- Code doesn't look for CWD redalert.(yml|yaml)
- Code doesn't look for $HOME/redalert.(yml|yaml)
- Code doesn't look for %%APPDATA%%/redalert.(yml|yaml)
- Code doesn't look for c:\redalert\tests.(yml|yaml) 
- Code doesn't look for /etc/redalert/redalert.(yml|yaml)